test/opt/loop_optimizations/unroll_simple.cpp

```diff
@@ -2998,6 +2998,68 @@ TEST_F(PassClassTest, OpPhiSelfReference) {
                                            kUnrollFactor);
 }
 
+// Test that a loop containing an unreachable merge block can still be unrolled
+// correctly.
+TEST_F(PassClassTest, UnreachableMerge) {
+  const std::string text = R"(
+; CHECK: OpSelectionMerge {{%\w+}} None
+; CHECK: OpSelectionMerge [[unrch1:%\w+]] None
+; CHECK: [[unrch1]] = OpLabel
+; CHECK-NEXT: OpUnreachable
+; CHECK: OpSelectionMerge {{%\w+}} None
+; CHECK: OpSelectionMerge [[unrch2:%\w+]] None
+; CHECK: [[unrch2]] = OpLabel
+; CHECK-NEXT: OpUnreachable
```

Comment:
I find the CHECKs easier to read if they're intermingled with the code.  I'm not sure how they relate to what the code is testing when they're all clustered above like this.  Is this way better for some other reason?

Reply:
In some cases, I see what you mean.  Personally, I find the check statements can get lost, even in mediums sized spir-v.  If I don't see them all together I miss them or find it harder to see how they relate.

In this case, I don't know a good way to get them to mingle with the code.  The first half of the check are for the first iteration of the loop, and the second are for the second iteration after it has been unrolled.  They correspond to the same statement in the original, but hey have to appear in this order.